src/Microsoft.IdentityModel.JsonWebTokens/JsonWebToken.cs

```diff
@@ -904,7 +906,11 @@ public string Subject
         /// <returns>Encoded token string without signature or authentication tag.</returns>
         public override string ToString()
         {
-            return EncodedToken.Substring(0, EncodedToken.LastIndexOf("."));
+            int lastDot = EncodedToken.LastIndexOf('.');
```

Comment:
>int lastDot = EncodedToken.LastIndexOf('.'); [](http://example.com/codeflow?start=12&length=44)

can we store lastDot and only recalculated when encodedtoken has been updated?

Reply:
As discussed, leaving as-is.

Reply:
I think we should change. We already have dot2 calculated when reading the token. additionally we also store IsSigned. the token may not be a JWS but may be a JWE and this code may not produce the desired outcome. Please see read token above for details.

Reply:
`JWT` -> you mean JWS in this case.

Reply:
it's better :) nit: from original comment "and only recalculated when encodedtoken has been updated"